src/main/java/org/broadinstitute/dsde/consent/ontology/datause/services/TextTranslationServiceImpl.java

```diff
@@ -255,49 +255,38 @@ private String findNamedClassType(Named n) {
             return "disease";
         }
 
-        OntClass commercial = model.getOntClass("http://www.broadinstitute.org/ontologies/DUOS/Commercial_Status");
-        if (cls.hasSuperClass(commercial)) {
+        OntClass duoDUR = model.getOntClass(DUO_DATA_USE_REQUIREMENTS);
+        if (cls.hasSuperClass(duoDUR)) {
             return "commercial";
         }
 
-        OntClass geography = model.getOntClass("http://www.broadinstitute.org/ontologies/DUOS/geography");
-        if (cls.hasSuperClass(geography)) {
-            return "geography";
-        }
-
-        OntClass population = model.getOntClass("http://www.broadinstitute.org/ontologies/DUOS/population");
-        if (cls.hasSuperClass(population)) {
-            return "population";
-        }
-
-        OntClass research_type = model.getOntClass("http://www.broadinstitute.org/ontologies/DUOS/research_type");
-        if (cls.hasSuperClass(research_type)) {
+        OntClass researchType = model.getOntClass(RESEARCH_TYPE);
+        OntClass duoSecondary = model.getOntClass(DUO_SECONDARY_CATEGORY);
+        if (cls.hasSuperClass(researchType) || cls.hasSuperClass(duoSecondary)) {
             return "research_type";
         }
 
-        OntClass dataset_usage = model.getOntClass("http://www.broadinstitute.org/ontologies/DUOS/dataset_usage");
-        if (cls.hasSuperClass(dataset_usage)) {
+        OntClass datasetUsage = model.getOntClass(DATASET_USAGE);
+        OntClass duoPrimary = model.getOntClass(DUO_PRIMARY_CATEGORY);
+        if (cls.hasSuperClass(datasetUsage) || cls.hasSuperClass(duoPrimary)) {
             return "dataset_usage";
         }
-        return null;
+
+        return "other";
```

Comment:
The switch from `null` seems strange to me... is it to avoid a NPE somewhere? I worry that some person or code might think that `http://www.broadinstitute.org/ontologies/DUOS/other` is a real thing.

Reply:
That is correct, there is the case where something could be null. In this case, we're not actually constructing a new term here, we're using the string to make a categorical decision about what kind of translation to make, disease, population, etc.